### PR TITLE
Refactor Azure packer builder

### PR DESCRIPTION
**Pull request by EleanorRigby**
- Templatize building for different OS with SIG and VHD
  - Arrange azure packer vars in lexical order

/assign @codenrhoden 
cc: @CecileRobertMichon

### CODE REVIEW

**Comment by codenrhoden** [comment]
This LGTM at first glance. Definitely want to hear from @CecileRobertMichon on this one. This change makes the way files are split up in line with AMI and OVA builders, and makes it much simpler to add additional OS's. There should be no discernable change in behavior when calling `make`. I still need to take this for a quick spin to confirm, though. Nice work!

**Comment by CecileRobertMichon** [comment]
I like it! Just need to make sure pipelines are still working with the change (I'll run a manual test since CI isn't there yet) but looks like a good improvement.

File: images/capi/packer/azure/packer.json [hunk 1]
```diff
@@ -112,49 +95,21 @@
...
   ],
   "provisioners": [
     {
       "type": "shell",
-      "only": ["vhd-ubuntu-1804", "sig-ubuntu-1804"],
+      "environment_vars": [
+        "BUILD_NAME={{user `build_name`}}"
+      ],
       "inline": [
+        "if [ $BUILD_NAME != \"ubuntu-1804\" ]; then exit 0; fi",
```
Comment (CecileRobertMichon) [inline images/capi/packer/azure/packer.json:112]: why is this needed? Why not just use `"only": ["vhd-ubuntu-1804", "sig-ubuntu-1804"],` since we know these are the two ubuntu-1804 builders?

**Comment by CecileRobertMichon** [comment]
Verified ubuntu pipelines are still running successfully with this change

File: images/capi/packer/azure/packer.json (hunk 1 above)
Comment (codenrhoden) [reply images/capi/packer/azure/packer.json:112]: This is working around a limitation in Packer. Unfortunately once you put a `user` var in the build name (e.g, in the builder section, instead of having `"name": "sig-ubuntu-1804`", we have `"name": "sig-{{user `build_name`}}",`, the `only` directives to packer no longer work, and you get errors about "build name not found". Sigh.

We had to do the same thing for AMIs: https://github.com/kubernetes-sigs/image-builder/blob/62d3c0e74b0650ebd62731a7cd558152c871c82c/images/capi/packer/ami/packer.json#L114-L118

**Comment by codenrhoden** [comment]
Awesome - a huge win for consistency!

/approve
